lib/grizzly/command_class/association_group_info.ex

```diff
@@ -0,0 +1,149 @@
+defmodule Grizzly.CommandClass.AssociationGroupInfo do
+  alias Grizzly.CommandClass.Mappings
+  require Logger
+
+  @type group_name_report :: %{group: non_neg_integer, name: String.t()}
+  @type groups_info_report :: %{groups_info: [group_info], dynamic: boolean}
+  @type group_info :: %{group: non_neg_integer, profile: group_profile}
+  @type group_profile :: atom
+  @type group_command_list_report :: %{group: non_neg_integer, commands: [group_command]}
+  @type group_command :: %{command_class: atom, command: atom}
+
+  @spec decode_group_name(non_neg_integer, binary) :: String.t()
+  def decode_group_name(length, binary) do
+    list = :erlang.binary_to_list(binary)
```

Comment:
Instead of making this a list to just check the length can we use

```
if byte_size(binary) != lenght do
```
And skip the transformation into a list?

Reply:
ah but of course